### ISSUE TITLE
panic: invalid argument to Int63n

### ISSUE THREAD

**Issue by gmugu** [rclone/rclone#7911]
#### What is the problem you are having with rclone?
Panic error occurred

#### What is your rclone version (output from `rclone version`)
v1.66.0

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Debian Arm64, Docker 

#### Which cloud storage system are you using? (e.g. Google Drive)
webdav

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
rclone serve dlna "dlna:/" --name=DLNA --dir-cache-time=0 --read-only --addr=":7879" --interface=eth0 --buffer-size=128M --header "Referer:"

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
alist-dlna-1    | 2024/06/08 20:39:50 NOTICE: webdav root '': Serving HTTP on 0.0.0.0:7879
alist-dlna-1    | 2024/06/08 20:39:50 NOTICE: DLNA server on :7879: Started SSDP on eth0
alist-dlna-1    | 2024/06/08 20:39:57 INFO  : /r/Xiaoya/%E7%94%B5%E5%BD%B1/%E4%B8%AD%E5%9B%BD/%E6%82%9F%E7%A9%BA%E4%BC%A0/%E6%82%9F%E7%A9%BA%E4%BC%A0.mkv: 192.168.3.42:60354 GET 206  
alist-dlna-1    | 2024/06/08 20:52:30 INFO  : /ctl: 192.168.3.42:60432 POST 200 "urn:schemas-upnp-org:service:ContentDirectory:1#Browse" 
alist-dlna-1    | 2024/06/08 20:52:34 INFO  : /ctl: 192.168.3.42:60432 POST 200 "urn:schemas-upnp-org:service:ContentDirectory:1#Browse" 
alist-dlna-1    | panic: invalid argument to Int63n
alist-dlna-1    | 
alist-dlna-1    | 
alist-dlna-1    | goroutine 3600 [running]:
alist-dlna-1    | math/rand.(*Rand).Int63n(0x1c02140?, 0x40007f03a8?)
alist-dlna-1    | 	/usr/local/go/src/math/rand/rand.go:122 +0xe4
alist-dlna-1    | math/rand.Int63n(0x0)
alist-dlna-1    | 	/usr/local/go/src/math/rand/rand.go:443 +0x28
alist-dlna-1    | github.com/anacrolix/dms/ssdp.(*Server).handle(0x4000242640, {0x40007d8000, 0x65, 0x5dc}, 0x4000710bd0)
alist-dlna-1    | 	/go/pkg/mod/github.com/anacrolix/dms@v1.6.0/ssdp/ssdp.go:319 +0x854
alist-dlna-1    | created by github.com/anacrolix/dms/ssdp.(*Server).serve in goroutine 39
alist-dlna-1    | 	/go/pkg/mod/github.com/anacrolix/dms@v1.6.0/ssdp/ssdp.go:129 +0x48
alist-dlna-1    | 2024/06/08 21:44:32 NOTICE: webdav root '': Serving HTTP on 0.0.0.0:7879
alist-dlna-1    | 2024/06/08 21:44:32 NOTICE: DLNA server on :7879: Started SSDP on eth0
alist-dlna-1    | panic: invalid argument to Int63n
alist-dlna-1    | 
alist-dlna-1    | 
alist-dlna-1    | goroutine 3155 [running]:
alist-dlna-1    | math/rand.(*Rand).Int63n(0x1c02140?, 0x400057e498?)
alist-dlna-1    | 	/usr/local/go/src/math/rand/rand.go:122 +0xe4
alist-dlna-1    | math/rand.Int63n(0x0)
alist-dlna-1    | 	/usr/local/go/src/math/rand/rand.go:443 +0x28
alist-dlna-1    | github.com/anacrolix/dms/ssdp.(*Server).handle(0x400098e3c0, {0x400056ac00, 0x65, 0x5dc}, 0x40007c1050)
alist-dlna-1    | 	/go/pkg/mod/github.com/anacrolix/dms@v1.6.0/ssdp/ssdp.go:319 +0x854
alist-dlna-1    | created by github.com/anacrolix/dms/ssdp.(*Server).serve in goroutine 42
alist-dlna-1    | 	/go/pkg/mod/github.com/anacrolix/dms@v1.6.0/ssdp/ssdp.go:129 +0x48
alist-dlna-1    | 2024/06/08 22:48:56 NOTICE: webdav root '': Serving HTTP on 0.0.0.0:7879
alist-dlna-1    | 2024/06/08 22:48:56 NOTICE: DLNA server on :7879: Started SSDP on eth0
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by albertony** [comment]
This looks like the following issue in third party component: https://github.com/anacrolix/dms/issues/123. It was fixed after its current latest release (v1.6.0), which is what rclone uses.

**Comment by ncw** [comment]
This was fixed upstream here: https://github.com/anacrolix/dms/commit/f4b323ae9c69b53df03216db7793da5ffb898dde

It looks like they haven't made a release recently though so I made an issue for that https://github.com/anacrolix/dms/issues/143

I pulled the master code down which has the fix in - please try this

[v1.67.0-beta.8032.5a5ce0540.fix-7911-serve-dlna](https://beta.rclone.org/branch/fix-7911-serve-dlna/v1.67.0-beta.8032.5a5ce0540.fix-7911-serve-dlna/) on branch [fix-7911-serve-dlna](https://github.com/rclone/rclone/tree/fix-7911-serve-dlna) (uploaded in 15-30 mins)

**Comment by gmugu** [comment]
Thank you for your reply. It has been solved.

**Comment by ncw** [comment]
Thank you for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.68 and if we make a v1.67.1 I'll put it in that.
